src/sig_gbdk.c

```diff
@@ -31,7 +31,34 @@ bool check_gbdk(void) {
 
     // GBDK 2.x - GBDK-2020 3.2.0
     if (CHECK_PATTERN_AT_ADDR(sig_gbdk_bmp, sig_gbdk_bmp_2x_to_2020_320_at)) {
-        entry_add_with_version(entry, STR_GBDK_2_x_to_2020_3_2_0);
+        // GBDK-2020 3.2.0
+        if (CHECK_PATTERN_AT_ADDR(sig_gbdk_0x164_GBDK_320, sig_gbdk_0x164_GBDK_320_at)) {
+            entry_add_with_version(entry, STR_GBDK_2020_3_2_0);
+            return true;
+        }
+
+        // GBDK 2.9.0 - GBDK-2020 3.1.0
+        if (CHECK_PATTERN_AT_ADDR(sig_gbdk_0x1c2_GBDK_29x, sig_gbdk_0x1c2_GBDK_29x_at)) {
+            // GBDK 2.9.5 - GBDK-2020 3.1.0
+            if (CHECK_PATTERN_AT_ADDR(sig_gbdk_0x1ca_GBDK_295, sig_gbdk_0x1ca_GBDK_295_at)) {
+                entry_add_with_version(entry, STR_GBDK_2_9_5_to_2020_3_1_0);
+                return true;
+            }
+
+            // GBDK 2.9.0 - GBDK 2.9.4
+            entry_add_with_version(entry, STR_GBDK_2_9_0_to_2_9_4);
+            return true;
+        }
+
+        // GBDK 2.0.18 - GBDK 2.0.15
+        if (CHECK_PATTERN_AT_ADDR(sig_gbdk_0x1c2_GBDK_2018_to_215, sig_gbdk_0x1c2_GBDK_2018_to_215_at)) {
+            entry_add_with_version(entry, STR_GBDK_2_0_18_to_2_1_5);
+            return true;
+        }
+
+        // GBDK 2.x, cannot narrow down further
+        // (for example, Demotronic uses a modified crt0 which fails the above checks)
+        entry_add_with_version(entry, STR_GBDK_2_x);
```

Comment:
Not too sure what to do about this here.

Reply:
That's ok, the same happens with modern GBDK-2020 programs that modify crt0. The modern ones, at least, aren't common and it just means they'll be detected as best they can.